package.json

```diff
@@ -53,8 +53,8 @@
   "husky": {
     "hooks": {
       "pre-commit": "lint-staged",
-      "pre-push": "yarn test",
-      "pre-publish": "yarn test"
+      "pre-push": "yarn test || npm test",
+      "pre-publish": "yarn test || npm test"
```

Comment:
If someone's making commits and pushes for this project without having Yarn installed, it means that they're maintaining an untracked `package-lock.json` file instead of updating the `yarn.lock` file in our version control, which isn't a good practice. Which package manager a project will use is a decision that everyone needs to stick with, we can't use both.

So unless there is a reason for this that I'm not seeing, I suggest reverting this.

Reply:
Oh, I see what the reason is, I just read #759 again. We have to make Travis use Yarn, that's something we should fix instead. Do you know why this is happening? Maybe something missing from `travis.yml`?

Reply:
As far [I can see...](https://travis-ci.org/yeoman/generator-webapp/jobs/540656581) is a problem related to our upgrade of dependencies.

```
yarn install v1.3.2
[1/5] Validating package.json...
[2/5] Resolving packages...
an-generator@^4.0.1an-generator@^4.0.1e@1.1.3ver@^5.6.0ocha@^6.1.4an-assert@^3.1.1bol-support-x@^1.4.1[3/5] Fetching packages...
error make-dir@3.0.0: The engine "node" is incompatible with this module. Expected version ">=8".
error Found incompatible module
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
The command "eval yarn " failed 3 times.
The command "yarn" failed and exited with 1 during .
```
One or more modules required node@8, this is why I suggested to remove support for node@6.

Reply:
I totally agree with losing Node v6, 👍 I was asking about the 2nd change, where you fall back to `npm test` if `yarn test` fails.

Reply:
Ah @silvenon, I think I got your point now! Well... I needed to add support for Npm as I am not using Yarn in this machine. As we are using Husky (i love it)... I wasn't able to push commits until `yarn test` pass... so I needed to add `npm test` for contributors/maintainers that are not using yarn (like myself) ;-)

Reply:
Is there a specific reason why you can't use it for this project? Imagine that you updated a package to version which has a crucial patch, I will still be stuck with the old version because `yarn.lock` stayed the same.

I used to try sticking with npm myself in Yarn projects, but it doesn't work as soon as you try to manage dependencies. Is not using Yarn a choice or a restriction for you?

Reply:
Because you do manage dependencies in #760, but not `yarn.lock`, which means that as soon as I pull in latest changes and run Yarn, I'll have to make a separate commit for updating the lockfile.

Reply:
I added the missing `yarn.lock` file also to #760 as https://github.com/yeoman/generator-webapp/pull/760/commits/0dc9aebc55d2593ebbe6d4ec97831cfabbf38670. I mostly use only NPM but I don't have problems to use Yarn as well ;-)

Reply:
Do you then agree that we should lose the ` || npm test` part because it gives the false impression that you can use npm too?

Reply:
Yes! Let's stick with Yarn.

Reply:
Added patch in https://github.com/yeoman/generator-webapp/pull/761/commits/e0e625d468bacc31f9a75fa5af5fbe8e2056b64d